fastify4ts/utils/controllerFactory.ts

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { routes, utils } from '@contrast/test-bench-utils';
+import { forEach } from 'lodash';
+import { FastifyInstance, FastifyRequest, FastifyReply } from 'fastify';
+
+/*
+ * Custom response functions allow you to change the functionality or return
+ * value of a sink endpoint.
+ *
+ * @callback ResponseFn
+ * @param {*}                result  return value of the sink method
+ * @param {fastify.Request}  request fastify request object
+ * @param {fastify.Response} reply   fastify reply object
+ */
+
+/*
+ * @type {ResponseFn}
+ */
+
+const defaultRespond = (result: unknown, request: FastifyRequest, reply: FastifyReply) => {
+  reply.type('text/html');
+  reply.send(result);
+};
+
+/*
+ * Configures a route to handle sinks configured by our shared test-bench-utils
+ * module.
+ *
+ * @param {string} vulnerability the vulnerability or rule being tested
+ * @param {Object} opts
+ * @param {Object} opts.locals additional locals to provide to EJS
+ */
+
+interface IFastifyInstance extends FastifyInstance {
+  [index: string]: any
+}
```

Comment:
I see why this was necessary, but it's really a problem with our underlying types in `test-bench-utils`. We shouldn't need this if the `method` param of the `sinkData` objects has the correct type.